system/jlib/jmetrics.cpp

```diff
@@ -304,3 +304,29 @@ void PeriodicMetricSink::doStopCollecting()
     collectThread.join();
     collectingHasStopped();
 }
+
+
+std::shared_ptr<CounterMetric> hpccMetrics::registerCounterMetric(const char *name, const char* desc, StatisticMeasure units, const MetricMetaData &metaData)
+{
+    return createMetricAndAddToManager<CounterMetric>(name, desc, units, metaData);
+}
+
+
+std::shared_ptr<GaugeMetric> hpccMetrics::registerGaugeMetric(const char *name, const char* desc, StatisticMeasure units, const MetricMetaData &metaData)
+{
+    return createMetricAndAddToManager<GaugeMetric>(name, desc, units, metaData);
+}
+
+
+std::shared_ptr<GaugeMetricFromCounters> hpccMetrics::registerGaugeFromCountersMetric(const char *name, const char* desc, StatisticMeasure units,
+                                                                         const std::shared_ptr<CounterMetric> &pBeginCounter, const std::shared_ptr<CounterMetric> &pEndCounter,
+                                                                         const MetricMetaData &metaData)
+{
+    //
+    // std::make_shared avoided to ensure separate memory allocations for the object and control structures. This ensures the control structure is not
```

Comment:
picky:  should be that it "ensures the vmt for the control structure does not live in the data belonging to the unloaded shared object."